--- /usr/local/lib/python3.7/dist-packages/boto-2.49.0-py3.7.egg/boto/connection.py	2020-01-16 15:57:46.058157070 +0000
+++ /tmp/connection.py	2020-01-16 15:58:49.377229234 +0000
@@ -793,18 +793,21 @@
         else:
             sock = socket.create_connection((self.proxy, int(self.proxy_port)))
         boto.log.debug("Proxy connection: CONNECT %s HTTP/1.0\r\n", host)
-        sock.sendall("CONNECT %s HTTP/1.0\r\n" % host)
-        sock.sendall("User-Agent: %s\r\n" % UserAgent)
+        sock.sendall(("CONNECT %s HTTP/1.0\r\n" % host).encode())
+        sock.sendall(("User-Agent: %s\r\n" % UserAgent).encode())
         if self.proxy_user and self.proxy_pass:
             for k, v in self.get_proxy_auth_header().items():
-                sock.sendall("%s: %s\r\n" % (k, v))
+                sock.sendall(("%s: %s\r\n" % (k, v)).encode())
             # See discussion about this config option at
             # https://groups.google.com/forum/?fromgroups#!topic/boto-dev/teenFvOq2Cc
             if config.getbool('Boto', 'send_crlf_after_proxy_auth_headers', False):
-                sock.sendall("\r\n")
+                sock.sendall(("\r\n").encode())
         else:
-            sock.sendall("\r\n")
-        resp = http_client.HTTPResponse(sock, strict=True, debuglevel=self.debug)
+            sock.sendall(("\r\n").encode())
+        kwargs = {'sock': sock, 'debuglevel': self.debug}
+        if six.PY2:
+            kwargs['strict'] = True
+        resp = http_client.HTTPResponse(**kwargs)
         resp.begin()
 
         if resp.status != 200:
